File: HGF_python-main/matlab/test_model_inversion.m

%% Test model inversion
load('test_cases/test_case_1_data.mat');
est = tapas_fitModel(data.y, data.input_u, ...
    'MS9_dmpad_hgf_ar1_lvl3_config', ...
    'MS9_dmpad_constant_voltemp_exp_config');


%% Test plotting
tapas_fit_plotCorr(est);
plot_hgf_binary_traj(est);


%% Test simulation
seed = 1;
sim = tapas_simModel(data.input_u,...
    'MS9_dmpad_hgf_ar1_lvl3', est.p_prc.p,...
    'MS9_dmpad_constant_voltemp_exp', est.p_obs.p, seed);
plot_hgf_binary_traj(sim);

seed = 2;
sim = tapas_simModel(data.input_u,...
    'MS9_dmpad_hgf_ar1_lvl3', est.p_prc.p,...
    'MS9_dmpad_constant_voltemp_exp', est.p_obs.p, seed);
plot_hgf_binary_traj(sim);

File: HGF_python-main/python/test_model_inversion.py
import scipy.io
from HGF.code_inversion.tapas_fitModel import tapas_fitModel
from HGF.code_inversion.tapas_quasinewton_optim_config import tapas_quasinewton_optim_config
from HGF.code_model1.prc_model.MS9_dmpad_hgf_ar1_lvl3_config import MS9_dmpad_hgf_ar1_lvl3_config
from HGF.code_model1.obs_model.MS9_dmpad_constant_voltemp_exp_config import MS9_dmpad_constant_voltemp_exp_config


# Load data
mat = scipy.io.loadmat('../test_cases/test_case_1_data.mat')
data = mat['data']

# Test model inversion
est = tapas_fitModel(data['y'][0][0], data['input_u'][0][0],
		c_prc = MS9_dmpad_hgf_ar1_lvl3_config,
		c_obs = MS9_dmpad_constant_voltemp_exp_config,
		c_opt = tapas_quasinewton_optim_config)
